java/fury-core/src/main/java/org/apache/fury/serializer/collection/ChildContainerSerializers.java

```diff
@@ -149,6 +149,14 @@ public Collection newCollection(MemoryBuffer buffer) {
       readAndSetFields(buffer, collection, slotsSerializers);
       return collection;
     }
+
+    @Override
+    public T copy(T originCollection) {
+      if (objectSerializer == null) {
+        objectSerializer = new ObjectSerializer<>(fury, type, false);
```

Comment:
Could we update `ObjectSerializer`:
```java
  public ObjectSerializer(Fury fury, Class<T> cls, boolean resolveParent) {
    super(fury, cls);
    // avoid recursive building serializers.
    // Use `setSerializerIfAbsent` to avoid overwriting existing serializer for class when used
    // as data serializer.
    if (resolveParent) {
      classResolver.setSerializerIfAbsent(cls, this);
    }
```